lib/sample_loader.rb

```diff
@@ -306,15 +306,26 @@ def load filename
     def samples_dir
       unless defined? @samples_dir
         @samples_dir = nil
-        base_dir = "#{File.dirname File.dirname __dir__}/"
-        cur_dir = Dir.getwd
-        if cur_dir.start_with? base_dir
-          gem_name = cur_dir[base_dir.length..].split("/").first
-          tentative_dir = File.join base_dir, gem_name, "samples"
-          @samples_dir = tentative_dir if ::File.directory? tentative_dir
-        end
+        pathname = Pathname.new Dir.getwd
+        gemspec_path = find_gemspec_path pathname
+        base_dir = File.dirname gemspec_path
+        tentative_dir = File.join base_dir, "samples"
+        @samples_dir = tentative_dir if ::File.directory? tentative_dir
       end
       @samples_dir
     end
+
+    ##
+    # Find the .gemspec file for the current library.
+    #
+    # @return [String] if the .gemspec file could be found.
+    # @return [nil] if the .gemspec file could not be found.
+    #
+    def find_gemspec_path pathname
+      pathname.ascend.each do |pn|
```

Comment:
I think `each` returns the original enumerable if the `break` is never executed. You probably need to figure out a way to return `nil` explicitly if gemspec is not found.

Reply:
I tested this. `gemspec_path` is `nil` if it doesn't find the file.

Reply:
Oh really. That's surprising to me, but okay, if it works, 👍🏼